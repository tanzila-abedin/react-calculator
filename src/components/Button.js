import '../style/App.css';
import PropTypes from 'prop-types';

const Button = ({ className, value, onClick }) => (
  <button
    type="button"
    data-testid="test-btn"
    className={className}
    onClick={onClick}
  >
    {value}
  </button>
);

Button.propTypes = {
  value: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};
export default Button;
