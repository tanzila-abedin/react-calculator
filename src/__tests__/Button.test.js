import { expect, test } from '@jest/globals';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import Button from '../components/Button';

test(' Snapshot of Button component', () => {
  const onClick = jest.fn();
  const tree = renderer
    .create(<Button onClick={onClick} value="AC" className="equals" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button component to display button value', async () => {
  const onClick = jest.fn();
  const buttonRender = render(
    <Button onClick={onClick} value="AC" className="equals" />,
  );
  const button = await buttonRender.findByTestId('test-btn');
  expect(button.innerHTML).toBe('AC');
});

test('Button component  to be able to click a button', async () => {
  const onClick = jest.fn();
  const buttonRender = render(
    <Button onClick={onClick} value="AC" className="equals" />,
  );
  const button = await buttonRender.findByTestId('test-btn');
  button.click();
  expect(onClick).toBeCalled();
});

test('Button component  returns a button element', async () => {
  const onClick = jest.fn();
  const buttonRender = render(
    <Button onClick={onClick} value="AC" className="equals" />,
  );
  const button = await buttonRender.findByTestId('test-btn');
  expect(button.type).toBe('button');
});
