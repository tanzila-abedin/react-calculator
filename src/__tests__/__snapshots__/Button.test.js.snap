// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Snapshot of Button component 1`] = `
<button
  className="equals"
  data-testid="test-btn"
  onClick={[MockFunction]}
  type="button"
>
  AC
</button>
`;
